### PR TITLE
change dna_hash to dna_address

### PR DESCRIPTION
**Pull request by thedavidmeister**
from #744 

## changes

- renames `dna_hash` to `dna_address`
- represents dna addresses as `Address` not `String` (it was never a `HashString` despite the name)
- implements `AddressableContent` for `Dna` so that `dna.address()` works (no more ad-hoc multihashing)
- implements feedback from parent issue

### CODE REVIEW

**Comment by Connoropolous** [comment]
@thedavidmeister do we still have the challenge where really big DNA entries can be passed over the network and written to the DHT, etc?

File: core/src/workflows/hold_link.rs [hunk 1]
```diff
@@ -76,10 +76,11 @@ pub mod tests {
             create_test_dna_with_wat("test_zome", "test_cap", Some(&test_wat_always_invalid()));
         dna.uuid = String::from("test_reject_invalid_link_on_hold_workflow");
 
-        // Hash of the original DNA
-        let dna_hash = base64::encode(&dna.multihash().unwrap());
+        // Address of the original DNA
+        let dna_address = dna.address();
```
Comment (Connoropolous) [inline core/src/workflows/hold_link.rs]: much better to have this consistent function than calling base64... repeatedly. Sweet!

File: net_connection/src/protocol_wrapper.rs [hunk 2]
```diff
@@ -34,13 +34,13 @@ pub struct ConfigData {
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
 pub struct ConnectData {
-    pub address: String,
+    pub address: Address,
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
 pub struct PeerData {
```
Comment (Connoropolous) [inline net_connection/src/protocol_wrapper.rs]: @thedavidmeister @neonphog perhaps a change on this side would require a change on the n3h side?

**Comment by thedavidmeister** [comment]
@Connoropolous this doesn't help anything with the size of DNA unfortunately, it's just tightening up the address/hash handling

File: core/src/workflows/hold_link.rs (hunk 1 above)
Comment (thedavidmeister) [reply core/src/workflows/hold_link.rs]: @Connoropolous yar, based on the experience where we decided Q3 last year to change the way addresses work for agents, i think it's pretty important to leverage `.address()` as much as possible - it keeps this as flexible as possible

File: net_connection/src/protocol_wrapper.rs (hunk 2 above)
Comment (thedavidmeister) [reply net_connection/src/protocol_wrapper.rs]: @Connoropolous i just chatted to @neonphog about these changes in teal, so he's aware, all good for merge i think :)

**Comment by Connoropolous** [comment]
thanks for the info David

**Pull request merged**